### PR TITLE
[DNM] Upmerge 04 11 2019

### PR DESCRIPTION
**Pull request by de-nordic**
Update to west.yml.

Update to Kconfig files that are related to changes, to Zephyr, done with pull requests: 
    https://github.com/zephyrproject-rtos/zephyr/pull/20281
    https://github.com/zephyrproject-rtos/zephyr/pull/20277

### CODE REVIEW

File: drivers/gps_sim/Kconfig [hunk 1]
```diff
@@ -68,7 +68,7 @@ config GPS_SIM_PSEUDO_RANDOM
 config GPS_SIM_DYNAMIC_VALUES
 	bool "GPS simulator dynamically creates data values"
 	default y
-	select NEWLIB_LIBC
+	select REQUIRES_FULL_LIBC
 	select NEWLIB_LIBC_FLOAT_PRINTF
```
Comment (de-nordic) [inline drivers/gps_sim/Kconfig]: Can anyone advise on whether this shall remain within Kconfig?

**Comment by de-nordic** [comment]
Depends on:
https://github.com/NordicPlayground/fw-nrfconnect-zephyr/pull/231
https://github.com/NordicPlayground/fw-nrfconnect-mbedtls/pull/5
https://github.com/NordicPlayground/fw-nrfconnect-mcuboot/pull/70

**Comment by pdunaj** [comment]
Hi @joerchan , do you know if anything happened in Zephyr that would caused problems with our LL?

**Comment by joerchan** [comment]
> Hi @joerchan , do you know if anything happened in Zephyr that would caused problems with our LL?

@pdunaj Nothing obvious that I can think of. Could you please give me a description of what is not working?

**Comment by pdunaj** [comment]
> @pdunaj Nothing obvious that I can think of. Could you please give me a description of what is not working?

Hard to say as I have not run on desk tests with this branch. According to CI expected connection is not set. I will check if I can do a quick test tomorrow to get more information.
Hi @wiminordic , is it possible to get the device logs?

**Comment by pdunaj** [comment]
I checked the FW now. I can connect and pair with dongle but following error happens after mouse reboots:
```
<err> settings: set-value failure. key: bt/ccc/f397a6136b0a1 error(-2)
```
The connection interval is also incorrectly negotiated (it looks as it was not negotiated).

**Comment by pdunaj** [comment]
I would really like to help more but this week and next we will be very busy. Following that we will be off for 2 weeks :(
Please make sure it works before merging. @wiminordic maybe you could run quick manual tests next week, after somebody fixes the BLE problems, before this is merged so we don't have devices corrupted.

**Comment by pdunaj** [comment]
BTW after this change building takes ages. I thought I am seeing things but I get 2.5s (master) vs 19s (PR). Somebody has broken something here?
@carlescufi

**Comment by mbolivar** [comment]
> BTW after this change building takes ages. I thought I am seeing things but I get 2.5s (master) vs 19s (PR). Somebody has broken something here?

I tried to reproduce on Linux with and without ccache enabled and can't reproduce. Build times between master and this PR are roughly the same for the nrf_desktop app.

**Comment by wiminordic** [comment]
> > @pdunaj Nothing obvious that I can think of. Could you please give me a description of what is not working?
> 
> Hard to say as I have not run on desk tests with this branch. According to CI expected connection is not set. I will check if I can do a quick test tomorrow to get more information.
> Hi @wiminordic , is it possible to get the device logs?

You can find the logs in Jenkins artifacts for the failing plan.
In the peripheral logs, on a re-connection attempt:
<wrn> ble_state: Security with e1:67:21:b0:28:1e (random) failed, level 1 err 4
<wrn> ble_state: Device disconnected
<err> bt_conn: not connected!

**Comment by de-nordic** [comment]
> 
> 
> > > @pdunaj Nothing obvious that I can think of. Could you please give me a description of what is not working?
> > 
> > 
> > Hard to say as I have not run on desk tests with this branch. According to CI expected connection is not set. I will check if I can do a quick test tomorrow to get more information.
> > Hi @wiminordic , is it possible to get the device logs?
> 
> You can find the logs in Jenkins artifacts for the failing plan.
> In the peripheral logs, on a re-connection attempt:
> `<wrn> ble_state: Security with e1:67:21:b0:28:1e (random) failed, level 1 err 4 <wrn> ble_state: Device disconnected <err> bt_conn: not connected!`

Is it possible that selection of random generator source/function does affect this?

**Comment by de-nordic** [comment]
> 
> 
> I checked the FW now. I can connect and pair with dongle but following error happens after mouse reboots:
> 
> ```
> <err> settings: set-value failure. key: bt/ccc/f397a6136b0a1 error(-2)
> ```
> 
> The connection interval is also incorrectly negotiated (it looks as it was not negotiated).
> 
> When I connect to my Android on the second identity mouse shows connection params being updated but now the cursor on Android does not move - i.e. there is no report transfer.
> 
> I see two errors now during reboot:
> 
> ```
> [00011133] <err> settings: set-value failure. key: bt/ccc/f397a6136b0a1 error(-2)
> [00011250] <err> settings: set-value failure. key: bt/ccc/8058f85619ce0/2 error(-2)
> ```

I'll look into changes done with integration of partition manager, they may have some impact on the storage of settings in flash.

**Comment by pdunaj** [comment]
> I'll look into changes done with integration of partition manager, they may have some impact on the storage of settings in flash.

So far problems with PM were visible when mcuboot was used. I have tested only application (no bootloader integration enabled).

**Comment by carlescufi** [comment]
@pdunaj @joerchan if we can fix this by disabling:

`BT_SETTINGS_CCC_LAZY_LOADING`
and
`BT_SMP_ALLOW_UNAUTH_OVERWRITE`

then let's do this for now, and then let's enable them separately and fix the code that goes with it.

**Comment by wiminordic** [comment]
> @pdunaj @joerchan @wiminordic if we can fix this by disabling:
> 
> `BT_SETTINGS_CCC_LAZY_LOADING`
> and
> `BT_SMP_ALLOW_UNAUTH_OVERWRITE`
> 
> then let's do this for now, and then let's enable them separately and fix the code that goes with it.

I checked on my desk with BT_SMP_ALLOW_UNAUTH_OVERWRITE and it seems to fix the issues on our side.

**Comment by pdunaj** [comment]
I created https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1522 to make sure fix for nrf_desktop enters repo before our trip. I will rebase this commit after that PR is merged.

**Comment by pdunaj** [comment]
please give me few more minutes for checking few more things

**Comment by pdunaj** [comment]
> please give me few more minutes for checking few more things

I am done. Quickly checked connections to Win10 and Android. Both working.

**Pull request merged**